File: Python_Library/Media.py
class Media:
	def __init__(self,a,b,c,d):
		self.call_number=a
		self.title=b
		self.subject=c
		self.notes=d
	
	def compare_call_number(self,x):
		found=self.call_number.find(x)
		if(found==-1):
			return False
		else:
			return True
			
	def compare_title(self,x):
		found=self.title.find(x)
		if(found==-1):
			return False
		else:
			return True
			
	def compare_subject(self,x):
		found=self.subject.find(x)
		if(found==-1):
			return False
		else:
			return True
	def compare_other(self,x):
		pass
		
	def display():
		pass
	
	

File: ObjectOrientedDesign(LibrarySearch)/Media.cpp
#include<iostream>
#include<string>
#include"Media.h"

using namespace std;

Media::Media(string a,string b,string c,string d)
{
 this->call_number=a;
 this->title=b;
 this->subject=c;
 this->notes=d;
}

bool Media::compare_title(const string& x) const
{
size_t found=this->title.find(x);
  if (found!=std::string::npos)
    return true;
  else
  return false;

}

bool Media::compare_call_number(const string& x) const
{
size_t found=this->call_number.find(x);
  if (found!=std::string::npos)
    return true;
  else
  return false;

}

bool Media::compare_subject(const string& x) const
{
size_t found=this->subject.find(x);
  if (found!=std::string::npos)
    return true;
  else
  return false;
}

bool Media::compare_other(const string& x) const
{
}

void Media::display()const
{
}

Media::~Media()
{
}
